nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,362 @@
+        return IngestClientFactory.createClient(ingestConnectionStringBuilder);
+    }
+
+    protected ManagedStreamingIngestClient createKustoStreamingIngestClient(final String clusterUrl,
+                                                                            final String appId,
+                                                                            final String appKey,
+                                                                            final String appTenant,
+                                                                            final KustoAuthenticationStrategy kustoAuthStrategy) throws URISyntaxException {
+        ConnectionStringBuilder ingestConnectionStringBuilder = createKustoEngineConnectionString(clusterUrl, appId, appKey, appTenant, kustoAuthStrategy);
+        ConnectionStringBuilder streamingConnectionStringBuilder = createKustoEngineConnectionString(clusterUrl, appId, appKey, appTenant, kustoAuthStrategy);
+        return IngestClientFactory.createManagedStreamingIngestClient(ingestConnectionStringBuilder, streamingConnectionStringBuilder);
+    }
+
+    public KustoIngestionResult ingestData(KustoIngestionRequest kustoIngestionRequest) throws URISyntaxException {
+        StreamSourceInfo info = new StreamSourceInfo(kustoIngestionRequest.getInputStream());
+        //ingest data
+        IngestionResult ingestionResult;
+        IngestionProperties ingestionProperties = new IngestionProperties(kustoIngestionRequest.getDatabaseName(),
+                kustoIngestionRequest.getTableName());
+
+        IngestionMapping.IngestionMappingKind ingestionMappingKind = setDataFormatAndMapping(kustoIngestionRequest.getDataFormat(), ingestionProperties);
+        if (StringUtils.isNotEmpty(kustoIngestionRequest.getMappingName()) && ingestionMappingKind != null) {
+            ingestionProperties.setIngestionMapping(kustoIngestionRequest.getMappingName(), ingestionMappingKind);
+        }
+
+        ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FAILURES_AND_SUCCESSES);
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE);
+        ingestionProperties.setFlushImmediately(false);
+        ingestionProperties.setIgnoreFirstRecord(StringUtils.equalsIgnoreCase(kustoIngestionRequest.getIgnoreFirstRecord(), "YES"));
+
+        try {
+            if (kustoIngestionRequest.isStreamingEnabled()) {
+                ingestionResult = managedStreamingIngestClient.ingestFromStream(info, ingestionProperties);
+            } else {
+                ingestionResult = queuedIngestClient.ingestFromStream(info, ingestionProperties);
+            }
+            if (kustoIngestionRequest.pollOnIngestionStatus()) {
+                List<IngestionStatus> statuses;
+                long startTime = System.currentTimeMillis();
+                long timeoutMillis = Long.parseLong(kustoIngestionRequest.getIngestionStatusPollingTimeout()) * 1000L;
```

Comment:
Instead of parsing the value in this method, the KustoIngestRequest should use a `java.time.Duration` and the value should be parsed in the Processor.